src/pages/pu-trx/[functionId]/PurchaseTransactionForm.js

```diff
@@ -0,0 +1,1662 @@
+                          formik.setFieldValue('header.tdPct', discount)
+                        } else {
+                          if (discount < 0 || formik.values.header.subtotal < discount) {
+                            discount = 0
+                          }
+                          formik.setFieldValue('header.tdAmount', discount)
+                        }
+                        formik.setFieldValue('header.currentDiscount', discount)
+                      }}
+                      onBlur={async e => {
+                        let discountAmount = Number(e.target.value.replace(/,/g, ''))
+                        let tdPct = Number(e.target.value.replace(/,/g, ''))
+                        let tdAmount = Number(e.target.value.replace(/,/g, ''))
+
+                        if (formik.values.header.tdType == DIRTYFIELD_TDPLAIN) {
+                          tdPct = (parseFloat(discountAmount) / parseFloat(subtotal)) * 100
+                          formik.setFieldValue('header.tdPct', tdPct)
+                        }
+
+                        if (formik.values.header.tdType == DIRTYFIELD_TDPCT) {
+                          tdAmount = (parseFloat(discountAmount) * parseFloat(subtotal)) / 100
+                          formik.setFieldValue('header.tdAmount', tdAmount)
+                        }
+                        setReCal(true)
+
+                        await recalcGridVat(formik.values.header.tdType, tdPct, tdAmount, discountAmount)
+                      }}
+                      onClear={() => {
+                        formik.setFieldValue('header.tdAmount', 0)
+                        formik.setFieldValue('header.tdPct', 0)
+                        recalcGridVat(formik.values.header.tdType, 0, 0, 0)
+                      }}
+                    />
+                    <CustomNumberField
+                      name='miscAmount'
+                      maxAccess={maxAccess}
+                      label={labels.misc}
+                      value={formik.values.header.miscAmount}
+                      decimalScale={2}
+                      readOnly={isPosted}
+                      onChange={e => formik.setFieldValue('header.miscAmount', e.target.value || 0)}
+                      onBlur={async () => {
+                        setReCal(true)
+                      }}
+                      onClear={() => formik.setFieldValue('header.miscAmount', 0)}
+                    />
+                    <CustomNumberField
+                      name='vatAmount'
+                      maxAccess={maxAccess}
+                      label={labels.VAT}
+                      value={vatAmount}
+                      readOnly
+                    />
+                    <CustomNumberField name='amount' maxAccess={maxAccess} label={labels.net} value={amount} readOnly />
+                  </Stack>
+                </Grid>
+              </Grid>
+            </Grid>
+          </Grid>
```

Comment:
try this for the fixed pard so you won't need to use inline css:

<Fixed>
          <Grid container spacing={2} sx={{ mt: '5px' }}>
            <Grid item xs={6}>
              <CustomTextArea
                name='description'
                label={labels.description}
                value={formik.values.header.description}
                rows={3}
                editMode={editMode}
                maxAccess={maxAccess}
                onChange={e => formik.setFieldValue('header.description', e.target.value)}
                onClear={() => formik.setFieldValue('header.description', '')}
                error={formik.touched.description && Boolean(formik.errors.description)}
              />
            </Grid>
            <Grid item xs={3}>
              <Stack spacing={2}>
                <CustomNumberField name='qty' maxAccess={maxAccess} label={labels.totQty} value={totalQty} readOnly />
                <CustomNumberField
                  name='volume'
                  maxAccess={maxAccess}
                  label={labels.totVolume}
                  value={totalVolume}
                  readOnly
                />
                <CustomNumberField
                  name='weight'
                  maxAccess={maxAccess}
                  label={labels.totWeight}
                  value={totalWeight}
                  readOnly
                />
              </Stack>
            </Grid>
            <Grid item xs={3}>
              <Stack spacing={2}>
                <CustomNumberField
                  name='subTotal'
                  maxAccess={maxAccess}
                  label={labels.subtotal}
                  value={subtotal}
                  readOnly
                />
                <CustomNumberField
                  name='tdAmount'
                  maxAccess={maxAccess}
                  label={labels.discount}
                  value={formik.values.header.tdAmount}
                  displayCycleButton={true}
                  cycleButtonLabel={cycleButtonState.text}
                  decimalScale={2}
                  readOnly={isPosted}
                  isPercentIcon={cycleButtonState.text === '%' ? true : false}
                  handleButtonClick={handleButtonClick}
                  ShowDiscountIcons={true}
                  onChange={e => {
                    let discount = Number(e.target.value.replace(/,/g, ''))
                    if (formik.values.header.tdType == DIRTYFIELD_TDPCT) {
                      if (discount < 0 || discount > 100) discount = 0
                      formik.setFieldValue('header.tdPct', discount)
                    } else {
                      if (discount < 0 || formik.values.header.subtotal < discount) {
                        discount = 0
                      }
                      formik.setFieldValue('header.tdAmount', discount)
                    }
                    formik.setFieldValue('header.currentDiscount', discount)
                  }}
                  onBlur={async e => {
                    let discountAmount = Number(e.target.value.replace(/,/g, ''))
                    let tdPct = Number(e.target.value.replace(/,/g, ''))
                    let tdAmount = Number(e.target.value.replace(/,/g, ''))

                    if (formik.values.header.tdType == DIRTYFIELD_TDPLAIN) {
                      tdPct = (parseFloat(discountAmount) / parseFloat(subtotal)) * 100
                      formik.setFieldValue('header.tdPct', tdPct)
                    }

                    if (formik.values.header.tdType == DIRTYFIELD_TDPCT) {
                      tdAmount = (parseFloat(discountAmount) * parseFloat(subtotal)) / 100
                      formik.setFieldValue('header.tdAmount', tdAmount)
                    }
                    setReCal(true)

                    await recalcGridVat(formik.values.header.tdType, tdPct, tdAmount, discountAmount)
                  }}
                  onClear={() => {
                    formik.setFieldValue('header.tdAmount', 0)
                    formik.setFieldValue('header.tdPct', 0)
                    recalcGridVat(formik.values.header.tdType, 0, 0, 0)
                  }}
                />
                <CustomNumberField
                  name='miscAmount'
                  maxAccess={maxAccess}
                  label={labels.misc}
                  value={formik.values.header.miscAmount}
                  decimalScale={2}
                  readOnly={isPosted}
                  onChange={e => formik.setFieldValue('header.miscAmount', e.target.value || 0)}
                  onBlur={async () => {
                    setReCal(true)
                  }}
                  onClear={() => formik.setFieldValue('header.miscAmount', 0)}
                />
                <CustomNumberField
                  name='vatAmount'
                  maxAccess={maxAccess}
                  label={labels.VAT}
                  value={vatAmount}
                  readOnly
                />
                <CustomNumberField name='amount' maxAccess={maxAccess} label={labels.net} value={amount} readOnly />
              </Stack>
            </Grid>
          </Grid>
        </Fixed>

Reply:
<Fixed>
          <Grid container spacing={2} sx={{ mt: '5px' }}>
            <Grid item xs={6}>
              <CustomTextArea
                name='description'
                label={labels.description}
                value={formik.values.header.description}
                rows={3}
                editMode={editMode}
                maxAccess={maxAccess}
                onChange={e => formik.setFieldValue('header.description', e.target.value)}
                onClear={() => formik.setFieldValue('header.description', '')}
                error={formik.touched.description && Boolean(formik.errors.description)}
              />
            </Grid>
            <Grid item xs={3}>
              <Stack spacing={2}>
                <CustomNumberField name='qty' maxAccess={maxAccess} label={labels.totQty} value={totalQty} readOnly />
                <CustomNumberField
                  name='volume'
                  maxAccess={maxAccess}
                  label={labels.totVolume}
                  value={totalVolume}
                  readOnly
                />
                <CustomNumberField
                  name='weight'
                  maxAccess={maxAccess}
                  label={labels.totWeight}
                  value={totalWeight}
                  readOnly
                />
              </Stack>
            </Grid>
            <Grid item xs={3}>
              <Stack spacing={2}>
                <CustomNumberField
                  name='subTotal'
                  maxAccess={maxAccess}
                  label={labels.subtotal}
                  value={subtotal}
                  readOnly
                />
                <CustomNumberField
                  name='tdAmount'
                  maxAccess={maxAccess}
                  label={labels.discount}
                  value={formik.values.header.tdAmount}
                  displayCycleButton={true}
                  cycleButtonLabel={cycleButtonState.text}
                  decimalScale={2}
                  readOnly={isPosted}
                  isPercentIcon={cycleButtonState.text === '%' ? true : false}
                  handleButtonClick={handleButtonClick}
                  ShowDiscountIcons={true}
                  onChange={e => {
                    let discount = Number(e.target.value.replace(/,/g, ''))
                    if (formik.values.header.tdType == DIRTYFIELD_TDPCT) {
                      if (discount < 0 || discount > 100) discount = 0
                      formik.setFieldValue('header.tdPct', discount)
                    } else {
                      if (discount < 0 || formik.values.header.subtotal < discount) {
                        discount = 0
                      }
                      formik.setFieldValue('header.tdAmount', discount)
                    }
                    formik.setFieldValue('header.currentDiscount', discount)
                  }}
                  onBlur={async e => {
                    let discountAmount = Number(e.target.value.replace(/,/g, ''))
                    let tdPct = Number(e.target.value.replace(/,/g, ''))
                    let tdAmount = Number(e.target.value.replace(/,/g, ''))

                    if (formik.values.header.tdType == DIRTYFIELD_TDPLAIN) {
                      tdPct = (parseFloat(discountAmount) / parseFloat(subtotal)) * 100
                      formik.setFieldValue('header.tdPct', tdPct)
                    }

                    if (formik.values.header.tdType == DIRTYFIELD_TDPCT) {
                      tdAmount = (parseFloat(discountAmount) * parseFloat(subtotal)) / 100
                      formik.setFieldValue('header.tdAmount', tdAmount)
                    }
                    setReCal(true)

                    await recalcGridVat(formik.values.header.tdType, tdPct, tdAmount, discountAmount)
                  }}
                  onClear={() => {
                    formik.setFieldValue('header.tdAmount', 0)
                    formik.setFieldValue('header.tdPct', 0)
                    recalcGridVat(formik.values.header.tdType, 0, 0, 0)
                  }}
                />
                <CustomNumberField
                  name='miscAmount'
                  maxAccess={maxAccess}
                  label={labels.misc}
                  value={formik.values.header.miscAmount}
                  decimalScale={2}
                  readOnly={isPosted}
                  onChange={e => formik.setFieldValue('header.miscAmount', e.target.value || 0)}
                  onBlur={async () => {
                    setReCal(true)
                  }}
                  onClear={() => formik.setFieldValue('header.miscAmount', 0)}
                />
                <CustomNumberField
                  name='vatAmount'
                  maxAccess={maxAccess}
                  label={labels.VAT}
                  value={vatAmount}
                  readOnly
                />
                <CustomNumberField name='amount' maxAccess={maxAccess} label={labels.net} value={amount} readOnly />
              </Stack>
            </Grid>
          </Grid>
        </Fixed>